### ISSUE TITLE
[JENKINS-42362] Investigate impact of parallel changes for declarative

### ISSUE THREAD

**Issue by jenkins-infra-bot** [jenkinsci/blueocean-plugin#3852]
<p>There is a proposal to clarify how to use parallel with declarative, using stages. This has the benefit of allowing config items that naturally live at stage level to be in parallel branches. This may result in some changes in flownodes/bismuth, and how it works with blue ocean visualisations (functionally things are similar): </p>

<p>A sample here: <a href="https://gist.github.com/abayer/bbe18e779696af88af9208f7502b898c" class="external-link" target="_blank" rel="nofollow noopener">https://gist.github.com/abayer/bbe18e779696af88af9208f7502b898c</a></p>

<p>The declarative ticket for this feature: <a href="https://issues.jenkins-ci.org/browse/JENKINS-41334" class="external-link" target="_blank" rel="nofollow noopener">https://issues.jenkins-ci.org/browse/JENKINS-41334</a><br/>
PR <a href="https://github.com/jenkinsci/pipeline-model-definition-plugin/pull/98" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/pipeline-model-definition-plugin/pull/98</a></p>

<p>In scope: </p>
<ul>
	<li>See what impact (there may be none) is of running declarative pipelines in this style with bismuth and blue ocean</li>
	<li>potentially make api adjustments/requests.</li>
</ul>

---
<details><summary><i>Originally reported by <img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/michaelneale.png" title="michaelneale's avatar" /> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=michaelneale">michaelneale</a>, imported from: <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-42362" target="_blank">Investigate impact of parallel changes for declarative</a></i></summary>
<i><ul>
<li><b>status</b>: Resolved
<li><b>priority</b>: Major
<li><b>component(s)</b>: blueocean-plugin
<li><b>resolution</b>: Duplicate
<li><b>resolved</b>: 2017-05-30T00:23:35+00:00
<li><b>votes</b>: 0
<li><b>watchers</b>: 2
<li><b>imported</b>: 2025-12-01
</ul></i>
<details><summary>Raw content of original issue</summary>

<pre>
<p>There is a proposal to clarify how to use parallel with declarative, using stages. This has the benefit of allowing config items that naturally live at stage level to be in parallel branches. This may result in some changes in flownodes/bismuth, and how it works with blue ocean visualisations (functionally things are similar): </p>

<p>A sample here: <a href="https://gist.github.com/abayer/bbe18e779696af88af9208f7502b898c" class="external-link" target="_blank" rel="nofollow noopener">https://gist.github.com/abayer/bbe18e779696af88af9208f7502b898c</a></p>

<p>The declarative ticket for this feature: <a href="https://issues.jenkins-ci.org/browse/JENKINS-41334" class="external-link" target="_blank" rel="nofollow noopener">https://issues.jenkins-ci.org/browse/JENKINS-41334</a>
PR <a href="https://github.com/jenkinsci/pipeline-model-definition-plugin/pull/98" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/pipeline-model-definition-plugin/pull/98</a></p>

<p>In scope: </p>
<ul>
	<li>See what impact (there may be none) is of running declarative pipelines in this style with bismuth and blue ocean</li>
	<li>potentially make api adjustments/requests.</li>
</ul>
</pre>
</details>
</details>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/jamesdumay.png" title="jamesdumay's avatar" /> jamesdumay:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-42362?focusedId=301231&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-301231">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Duplicate of <a href="https://issues.jenkins.io/browse/JENKINS-44391" title="Support for visualizing parallel stages" class="issue-link" data-issue-key="JENKINS-44391"><del>JENKINS-44391</del></a></p>
</pre>
</details>

</ul>
</details>

<p>Duplicate of <a href="https://issues.jenkins.io/browse/JENKINS-44391" title="Support for visualizing parallel stages" class="issue-link" data-issue-key="JENKINS-44391"><del>JENKINS-44391</del></a></p>

**Comment by jenkins-infra-bot** [comment]
<i>[Original `duplicates` from Jira: <a href="https://github.com/jenkinsci/blueocean-plugin/issues?q=is%3Aissue%20%22jira_issue_key%3DJENKINS-44391%22">JENKINS-44391</a>]</i>

**Comment by jenkins-infra-bot** [comment]
<i>[Original `blocks` from Jira: <a href="https://github.com/jenkinsci/blueocean-plugin/issues?q=is%3Aissue%20%22jira_issue_key%3DJENKINS-41334%22">JENKINS-41334</a>]</i>
